### PR TITLE
Js tracking of k2-site examples

### PR DESCRIPTION
**Pull request by nlazarov**
Tracking of JavaScript exceptions is done through the [Sentry](http://sentry.io/) service. I am using my email for the trial period of the tracking. If all is well, we should decide on a corporate level email.

### CODE REVIEW

File: docs-public/snippets.js [hunk 1]
```diff
@@ -282,10 +282,20 @@ function bootstrapAngular(code, resize) {
         code,
         "import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';",
         "import { NgModule } from '@angular/core';",
+        "import Raven = require('raven-js');",
+        "import { ErrorHandler } from '@angular/core';",
+        "console.log(Raven, Raven.config);",
```
Comment (gyoshev) [inline docs-public/snippets.js]: Leftover `console.log`.

Also, the imports from `@angular/core` can be merged.
`import { NgModule, ErrorHandler } from '@angular/core'`

File: docs-public/example-runner.js [hunk 2]
```diff
@@ -106,6 +109,9 @@ var ExampleRunner = (function() {
             });
 
             system.config({
+                paths: {
+                    'raven-js': npmUrl + "/raven-js"
+                },
```
Comment (gyoshev) [inline docs-public/example-runner.js]: Is this necessary? All other modules seem to be functional without declaring paths.

Comment (nlazarov) [inline docs-public/example-runner.js]: You are right - will remove it.

**Comment by nlazarov** [comment]
I should also add a flag for enabling the JS tracking from _config.yml. So, please, don't merge yet.

File: docs-public/snippets.js (hunk 1 above)
Comment (nlazarov) [inline docs-public/snippets.js]: Ah, shame on me.

**Comment by nlazarov** [comment]
Added the JS tracking toggle option.

**Comment by tsvetomir** [comment]
Does this change mean that we'll get error notifications for examples that were modified by the user?

File: docs-layout.hbs [hunk 3]
```diff
@@ -30,6 +30,7 @@
       var runnerScript = '/internals/example-runner.js';
       var plunkerBluePrintPath = '/internals/plunker/';
       var npmUrl = '/npm';
+      var trackjs = {{trackjs}};
```
Comment (gyoshev) [inline docs-layout.hbs]: This results in a syntax error (`var trackjs = ;`) in projects that run `gulp docs`. You can test this in e.g. kendo-angualar-dialog, after linking the common-tasks repo.

Comment (nlazarov) [inline docs-layout.hbs]: I will revise this part.

**Comment by nlazarov** [comment]
Fixed the docs-layout.hbs problem. I also added the JS tracking in the home page demos.

**Comment by gyoshev** [comment]
@nlazarov all seems well. Can you please rebase and:

- squash any excessive commits
- reword one of the commits as a feature, so that a new package version is released upon merging

**Comment by nlazarov** [comment]
Commits are squashed.

**Pull request merged**